core/src/main/java/com/datastrato/gravitino/meta/TagEntity.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+
+package com.datastrato.gravitino.meta;
+
+import com.datastrato.gravitino.Audit;
+import com.datastrato.gravitino.Auditable;
+import com.datastrato.gravitino.Entity;
+import com.datastrato.gravitino.Field;
+import com.datastrato.gravitino.HasIdentifier;
+import com.datastrato.gravitino.MetadataObject;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.tag.Tag;
+import com.google.common.collect.Maps;
+import java.util.Collections;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+
+public class TagEntity implements Tag, Entity, Auditable, HasIdentifier {
+
+  public static final Field ID =
+      Field.required("id", Long.class, "The unique id of the tag entity.");
+
+  public static final Field NAME =
+      Field.required("name", String.class, "The name of the tag entity.");
+
+  public static final Field COMMENT =
+      Field.optional("comment", String.class, "The comment of the tag entity.");
+
+  public static final Field PROPERTIES =
+      Field.optional("properties", Map.class, "The properties of the tag entity.");
+
+  public static final Field ASSOCIATED_OBJECTS =
+      Field.optional(
+          "objects", MetadataObject[].class, "The associated objects of the tag entity.");
+
+  public static final Field AUDIT_INFO =
+      Field.required("audit_info", Audit.class, "The audit details of the tag entity.");
+
+  private Long id;
+  private String name;
+  private Namespace namespace;
+  private String comment;
+  private Map<String, String> properties;
+  private MetadataObject[] objects = null;
```

Comment:
This maintains the mapping of tags to entities. Where do we keep the reverse mapping?

Reply:
We don't have to maintain this in memory, we can use the method to get tags for a specific metadata object, for example, like `listTagsForMetadataObject(String metalake, MetadataObject metadataObject)` here I defined in `TagManager`.